### DISCUSSION TITLE
I'm looking forward to rewriting i2pd in Rust/Zig

### DISCUSSION THREAD

**Discussion by ghost** [PurpleI2P/i2pd#2103] [Ideas]
https://github.com/ziglang/zig
Zig is smaller and faster than Rust

**Comment by Vort** [comment]
C++ language of i2pd won't change.
But you may make your own fork of course.
However, i2pd is complex, so rewriting won't be easy.

**Reply by x64x2** [reply]
make i2pd less bloat and so simple

**Reply by Vort** [reply]
Simplification is even harder task than just rewriting.

**Comment by anikey-from-i2p** [comment]
For Rust you may look at the ire project: https://github.com/str4d/ire

I haven't heard of a Zig i2p daemon project.

I think if you want another language, you need to create another project, not rewrite the existing one.
